### PR TITLE
filter out mixins for mods that are not loaded

### LINKED ISSUE #59: @Pseudo annotation logs errors on startup if class cannot be found
Can be confusing for users, see LordDeatHunter/HarvestScythes#14

### PR DESCRIPTION
**Pull request by trobol**
Resolves #59 

The mixin plugin idea comes from https://github.com/DimensionalDevelopment/VanillaFix

### CODE REVIEW

**Comment by P03W** [comment]
This was originally a mini plugin, but I switched to @Pseudo so that I would have to deal with additional files, is there some advantage I'm not aware of?

**Comment by trobol** [comment]
The problem with @pseudo is that Mixin tries to load the class and is what spits out the big stack trace.
I originally thought it was a bug in fabric but its actually intentionally part of Mixin https://github.com/FabricMC/fabric-loader/issues/543

The only way to get around having the warning is to prevent Mixin from trying to load the class that does not exist.

The way I'm doing that here is using the Mixin plugin to filter out some of the classes we are trying to inject when the mod isn't loaded.
 
TLDR: No more big warning when missing optional mods

**Comment by P03W** [comment]
Big stack trace? Ive only ever seen a single line though, did something change recently that's making this a significant issue?

**Comment by trobol** [comment]
I am working with 1.18 so it could be something with that or it might be my logging level. I'll look into it.

<details>

<summary> The warning I was getting (theres also one for harvest_scythes but its basically identical)</summary>

```
[14:00:55] [main/WARN]: Catching java.lang.ClassNotFoundException: info/tehnut/harvest/Harvest
java.lang.ClassNotFoundException: info/tehnut/harvest/Harvest
at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.getClassBytes(MixinServiceKnot.java:65) ~[fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.getClassNode(MixinServiceKnot.java:76) ~[fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.getClassNode(MixinServiceKnot.java:71) ~[fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.ClassInfo.forName(ClassInfo.java:2022) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinInfo.getTargetClass(MixinInfo.java:1018) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinInfo.readTargetClasses(MixinInfo.java:1008) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinInfo.parseTargets(MixinInfo.java:896) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:869) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:777) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:539) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) [fabric-server-launch.jar:?]
at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:222) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:133) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) [fabric-server-launch.jar:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:519) [?:?]
at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:566) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-server-launch.jar:?]
at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:72) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:98) [fabric-server-launch.jar:?]
at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:60) [fabric-server-launch.jar:?]
```
</details>


However, regardless the length of the warning, it is confusing and I'm not sure saving that extra file is worth it.

**Comment by sfPlayer1** [comment]
We don't officially support Mixin plugins due to their large API surface, so my solution is essentially the only option that can be advertised, even if it is rather cumbersome ;) You are ofc perfectly fine with the plugin solution for now.
